main/src/cgeo/geocaching/EditWaypointActivity.java

```diff
@@ -539,6 +539,20 @@ protected Void doInBackground(final Void... params) {
                     finishHandler.sendEmptyMessage(SAVE_ERROR);
                     return null;
                 }
+                if (own) {
+                    int max = 0;
+                    for (final Waypoint w : cache.getWaypoints()) {
+                        if (w.getPrefix().startsWith("OWN")) {
+                            try {
+                                final int suffix = Integer.parseInt(w.getPrefix().substring(3));
+                                if (suffix > max) max = suffix;
+                            } catch(final NumberFormatException ignored) {
+
+                            }
+                        }
+                    }
+                    waypoint.setPrefix(waypoint.getPrefix()+String.valueOf(max+1));
+                }
```

Comment:
While thinking about this implementation I also scanned our code and cross-checked with gc.com, in order to better understand what we should be doing here.
First of all is the prefix meant to have only to digits or letters, as it replaces the 'GC' part in the geocode when creating a waypoint name, especially for gpx export (PKXXXXX, S1XXXXX) (was not aware of that either).
Therefore already code existed, that was meant to make prefixes unique within the cache (and adhere to the length limit), see GpxSerializer.java,202ff and Geocache.java,1383ff and 1350ff
This was 'damaged' when fixing #4686 but I think the fix to that fix should also adhere to the conceptual limits of the prefix (so cross-checking gpx ex- and import seems a good idea)
On a side-note is there also a gap in this implementation, as you compare the prefix of the other waypoints with "own" (we have a constant for that), but then use the previous prefix to construct the new one (implicitly assuming that "own" was assigned here when constructing the waypoint)